ssv/committee_runner.go

```diff
@@ -0,0 +1,506 @@
+	return cr.valCheck
+}
+
+func (cr CommitteeRunner) GetNetwork() Network {
+	return cr.network
+}
+
+func (cr CommitteeRunner) HasRunningDuty() bool {
+	return cr.BaseRunner.hasRunningDuty()
+}
+
+func (cr CommitteeRunner) ProcessPreConsensus(signedMsg *types.PartialSignatureMessages) error {
+	return errors.New("no pre consensus phase for committee runner")
+}
+
+func (cr CommitteeRunner) ProcessConsensus(msg *types.SignedSSVMessage) error {
+	decided, decidedValue, err := cr.BaseRunner.baseConsensusMsgProcessing(cr, msg)
+	if err != nil {
+		return errors.Wrap(err, "failed processing consensus message")
+	}
+
+	// Decided returns true only once so if it is true it must be for the current running instance
+	if !decided {
+		return nil
+	}
+
+	duty := cr.BaseRunner.State.StartingDuty
+	postConsensusMsg := &types.PartialSignatureMessages{
+		Type:     types.PostConsensusPartialSig,
+		Slot:     duty.DutySlot(),
+		Messages: []*types.PartialSignatureMessage{},
+	}
+
+	beaconVote := decidedValue.(*types.BeaconVote)
+	for _, duty := range duty.(*types.CommitteeDuty).BeaconDuties {
+		switch duty.Type {
+		case types.BNRoleAttester:
+			attestationData := constructAttestationData(beaconVote, duty)
+			err = cr.GetSigner().IsAttestationSlashable(cr.GetBaseRunner().Share[duty.ValidatorIndex].SharePubKey,
+				attestationData)
+			if err != nil {
+				return errors.Wrap(err, "attempting to sign slashable attestation data")
+			}
+			partialMsg, err := cr.BaseRunner.signBeaconObject(cr, duty, attestationData, duty.DutySlot(),
+				types.DomainAttester)
+			if err != nil {
+				return errors.Wrap(err, "failed signing attestation data")
+			}
+			postConsensusMsg.Messages = append(postConsensusMsg.Messages, partialMsg)
+
+		case types.BNRoleSyncCommittee:
+			blockRoot := beaconVote.BlockRoot
+			partialMsg, err := cr.BaseRunner.signBeaconObject(cr, duty, types.SSZBytes(blockRoot[:]), duty.DutySlot(),
+				types.DomainSyncCommittee)
+			if err != nil {
+				return errors.Wrap(err, "failed signing sync committee message")
+			}
+			postConsensusMsg.Messages = append(postConsensusMsg.Messages, partialMsg)
+		}
+	}
+
+	ssvMsg := &types.SSVMessage{
+		MsgType: types.SSVPartialSignatureMsgType,
+		MsgID:   types.MessageID(cr.GetBaseRunner().QBFTController.Identifier),
+	}
+	ssvMsg.Data, err = postConsensusMsg.Encode()
+	if err != nil {
+		return errors.Wrap(err, "failed to encode post consensus signature msg")
+	}
+
+	msgToBroadcast, err := types.SSVMessageToSignedSSVMessage(ssvMsg, cr.BaseRunner.QBFTController.CommitteeMember.OperatorID,
+		cr.operatorSigner.SignSSVMessage)
+	if err != nil {
+		return errors.Wrap(err, "could not create SignedSSVMessage from SSVMessage")
+	}
+
+	if err := cr.GetNetwork().Broadcast(ssvMsg.MsgID, msgToBroadcast); err != nil {
+		return errors.Wrap(err, "can't broadcast partial post consensus sig")
+	}
+	return nil
+
+}
+
+func (cr CommitteeRunner) ProcessPostConsensus(signedMsg *types.PartialSignatureMessages) error {
```

Comment:
spec tested?

Reply:
In `CommitteeSpecTest`, e.g. in `SequencedHappyFlowDuties`